book/src/proptest-derive/errors.md

```diff
@@ -0,0 +1,631 @@
+
+- Pass nothing at all. E.g., `#[proptest(value)]`.
+
+- Use another illegal form. E.g., `#[proptest(value("a", "b"))]`.
+
+- Pass a string expression which is not valid Rust syntax, as in the above
+  example.
+
+## E0027
+
+This error happens if an invalid form of `#[proptest(filter..)]` is used.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    #[proptest(filter = "> 3")] // String content does not name a function
+    g1: u128,
+}
+```
+
+There are a few different ways to get this error:
+
+- Pass nothing at all. E.g., `#[proptest(filter)]`.
+
+- Use another illegal form. E.g., `#[proptest(filter("a", "b"))]`.
+
+- Pass a string expression which is not valid Rust syntax, as in the above
+  example.
+
+## E0028
+
+This error occurs if a modifier which implies a value is to be generated is
+applied to an enum variant which is also marked `#[proptest(skip)]`.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+enum Enum {
+    V1(u32),
+    #[proptest(skip, value = "Enum::V2(42)")]
+    V2(u32),
+}
+```
+
+Here, the `#[proptest(value)]` modifier suggests the user intends some value to
+be generated for the enum variant, but at the same time `#[proptest(skip)]`
+indicates not to generate that variant.
+
+## E0029
+
+This error happens if a modifier which would constrain or control how the value
+of an enum variant is to be generated is applied to a unit variant.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+enum Foo {
+    #[proptest(value = "Foo::V1")]
+    V1,
+    // ...
+}
+```
+
+Unit variants only have one possible value, so there is only one possible
+strategy. As a result, it is pointless to try to specify an alternate strategy
+or to filter such variants.
+
+## E0030
+
+This error happens if a modifier which would constrain or control how the value
+of a struct is to be generated is applied to a unit struct.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(params = "u8")]
+struct Foo;
+```
+
+Unit structs only have one possible value, so there is only one possible
+strategy. As a result, it is pointless to try to specify an alternate strategy
+or to filter such structs.
+
+## E0031
+
+This error occurs if `#[proptest(no_bound)]` is applied to something that is
+not a type variable.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    #[proptest(no_bound)]
+    bar: u32,
+}
+```
+
+The `no_bound` modifier only makes sense on generic type variables, as in
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo<#[proptest(no_bound)] T> {
+    #[proptest(value = "None")]
+    bar: Option<T>,
+}
+```
+
+## E0032
+
+This error happens if `#[proptest(no_bound)]` is passed anything.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo<#[proptest(no_bound = "yes")] T> {
+    _bar: PhantomData<T>,
+}
+```
+
+The only valid form for the modifier is `#[proptest(no_bound)]`.
+
+## E0033
+
+This error occurs if the sum of the weights on the variants of an enum overflow
+a `u32`.
```

Comment:
```suggestion
an `u32`.
```

Reply:
How do you pronounce it? I say "you thirty two" which definitely takes "a".

Reply:
Interesting; I say it the same way so you are right here. :)